src/Wellcome.Dds/Wellcome.Dds.AssetDomainRepositories/Ingest/CloudServicesIngestRegistry.cs

```diff
@@ -0,0 +1,179 @@
+        public DlcsIngestJob GetJob(int id)
+        {
+            return ddsInstrumentationContext.DlcsIngestJobs
+                .Include(j => j.DlcsBatches)
+                .FirstOrDefault(j => j.Id == id);
+        }
+
+        public IEnumerable<DlcsIngestJob> GetQueue(DateTime? after)
+        {
+            var query = ddsInstrumentationContext.DlcsIngestJobs
+                .OrderByDescending(j => j.Created)
+                .Include(j => j.DlcsBatches)
+                .Where(j => j.StartProcessed == null);
+
+            if (after.HasValue)
+            {
+                return query.Where(j => j.Created > after.Value).ToArray();
+            }
+            return query.ToArray();
+        }
+
+        public IEnumerable<DlcsIngestJob> GetProblems(DateTime? after)
+        {
+            // .NET Core migration note.
+            // I have removed the BatchWithJob class as it was only used internally to materialise 
+            // query results from the sqlServer query below... this method still just returns
+            // IEnumerable<DlcsIngestJob>
+            // So we need to re-write the way it gets jobs with problems.
+            // 
+            // Most recent 100 DlcsIngestJobs 
+            // that have Succeeded=0, or have a batch with ErrorText not null
+            // where batches have ContentLength > 0
+
+            var problems = ddsInstrumentationContext.DlcsIngestJobs
+                .Where(j => j.DlcsBatches.All(b => b.ContentLength > 0))
+                .Where(j => !j.Succeeded || j.DlcsBatches.Any(b => b.ErrorText != null))
+                .OrderByDescending(j => j.Id)
+                .Take(100);
+
+            return problems.ToArray();
+
+            // Please see 
+            // https://github.com/wellcomelibrary/dds-ecosystem/blob/new-storage-service/wellcome-dds/Wellcome.Dds/Ingest/CloudServicesIngestRegistry.cs#L78
+            // for comparison         
+        }
+
+        private DlcsIngestJob NewJob(string identifier, int sequenceIndex, string volumeIdentifier,
+            string issueIdentifier, bool useInitialOrigin, bool immediateStart)
+        {
+            var job = new DlcsIngestJob
+            {
+                Created = DateTime.Now,
+                Identifier = identifier,
+                SequenceIndex = sequenceIndex,
+                VolumePart = volumeIdentifier,
+                IssuePart = issueIdentifier
+            };
+
+            if (useInitialOrigin)
+            {
+                job.Data = "initialOrigin";
+            }
+            if (immediateStart)
+            {
+                job.StartProcessed = DateTime.Now;
+            }
+            return job;
+        }
+
+        // TODO: Making this async needs a code review (compare CloudServicesIngestRegistry and its interface with dds-ecosystem)
+        // The following few mthods all need reviewing
+        public async IAsyncEnumerable<DlcsIngestJob> RegisterImagesForImmediateStartAsync(string identifier)
```

Comment:
Can this become 

```cs
public IAsyncEnumerable<DlcsIngestJob> RegisterImagesForImmediateStartAsync(string identifier) 
  => RegisterImagesInternalAsync(identifier, false, true);
```

Reply:
"Cannot return a value from an iterator. Use the yield return statement... " is the error I get when I try to do that.